### PR TITLE
Use `Out` instead of `resCellX`

### PR DESCRIPTION
**Pull request by jeremyrsmith**
Introduces a bunch of things to make the UI cleaner in general.

- Get rid of `resCellX`. It's now replaced with `Out`.
- UI shows `Out:` next to the `Out` value, if there is one. Declared values are not shown in the output, but are highlighted in the symbol table.

### CODE REVIEW

**Comment by jeremyrsmith** [comment]
For context, now looks like this:
screen shot 2019-02-26 at 1 45 19 pm

File: polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaInterpreter.scala [hunk 1]
```diff
@@ -67,6 +67,9 @@ class ScalaInterpreter(
   ): IO[Stream[IO, Result]] = {
     val originalOut = System.out
 
+    // TODO: this is wrong. It doesn't seem good to keep parsing this from "CellXX" though. Maybe cells also (or only?) need a numeric ID?
```
Comment (indig) [inline polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaInterpreter.scala]: Agreed, I have also been thinking that cells need their own ID space thats separate from position in the notebook and execution order.

**Comment by jeremyrsmith** [comment]
Now looks like this:
screen shot 2019-02-26 at 4 00 00 pm

File: polynote-frontend/style/styles.less [hunk 2]
```diff
@@ -296,7 +296,7 @@ body {
 
     .cell-input-editor {
       border: 1px solid @ui-border;
-      border-left-width: 0;
+      border-left: 2pt solid transparent;
```
Comment (indig) [inline polynote-frontend/style/styles.less]: Just curious - any reason to use a border vs a margin or padding? I've never had a good grasp of when to use which. (maybe it doesn't make a difference?)

File: polynote-frontend/polynote/cell.js [hunk 3]
```diff
@@ -350,17 +362,36 @@ export class CodeCell extends Cell {
...
             self.cellOutputDisplay.appendChild(
                 div(['output'], err)
             );
         });
     }
 
+    addResult(result) {
+        if (result instanceof ResultValue) {
+            // TODO: keep "result" and "output" separate for UI... have a way to show declarations, results, outputs, etc. separately
+            if (result.name !== "Out") {
```
Comment (indig) [inline polynote-frontend/polynote/cell.js]: If a user declares a `val Out = ...` will it show up here?

If so, maybe that's not necessarily a bad thing - maybe I want to hook into the automatic repr display for `Out`s for a value that for some reason can't be the return value of the cell...?

File: polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala [hunk 4]
```diff
@@ -173,7 +173,7 @@ class PythonInterpreter(val symbolTable: RuntimeSymbolTable) extends LanguageInt
         jep.eval("__polynote_last__ = __polynote_parsed__[-1]")
         val lastIsExpr = jep.getValue("isinstance(__polynote_last__, ast.Expr)", classOf[java.lang.Boolean])
 
-        val resultName = s"res$cell"
+        val resultName = "Out"
```
Comment (indig) [inline polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala]: Maybe `"Out"` should be a LanguageInterpreter-level constant?

File: polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaInterpreter.scala [hunk 5]
```diff
@@ -150,19 +153,11 @@ class ScalaInterpreter(
                       runtimeTools.eval(tree) -> typ
                     }
                     val methodType = importFromRuntime.importType(fnType)
-                    Some(ResultValue(kernelContext)(method.name.toString, methodType, runtimeFn, cell))
-
-                }.flatten
 
-                val maybeOutput = syms.find(_.name.startsWith("res"))
+                    // I guess we're saying a "def" shouldn't become the cell result?
```
Comment (indig) [inline polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaInterpreter.scala]: I think that makes sense, given that a def will already have a name associated with it

**Comment by indig** [comment]
@jeremyrsmith the build failure looks legit, seems like the generator found a super edge case in ContentEditSpec?

**Comment by jeremyrsmith** [comment]
@indig yeah, I am going to merge this but I added a TODO to explore that test case.

File: polynote-frontend/style/styles.less (hunk 2 above)
Comment (jeremyrsmith) [reply polynote-frontend/style/styles.less]: I used border because it already has a margin-left 😛

File: polynote-frontend/polynote/cell.js (hunk 3 above)
Comment (jeremyrsmith) [reply polynote-frontend/polynote/cell.js]: Yeah, nothing stopping you from declaring `val Out`... but then if you also have an expression as the last line, it would be a compile error for redeclaring that. Not super worried about it right now, but maybe that's a case we should detect (and then don't capture the expression, sorry 'bout it)

File: polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala (hunk 4 above)
Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala]: :+1: agreed
